src/GafferOSL/OSLObject.cpp

```diff
@@ -310,9 +316,23 @@ void OSLObject::hashProcessedObject( const ScenePath &path, const Gaffer::Contex
 	{
 		h.append( inPlug()->fullTransformHash( path ) );
 	}
-	if( useAttributesPlug()->getValue() )
+
+	if( gafferAttributes )
 	{
-		h.append( inPlug()->fullAttributesHash( path ) );
+		for( const auto &i : gafferAttributes->members() )
```

Comment:
I think we should adopt structured bindings as our standard idiom for this sort of loop, instead of using `i.first` and `i.second` :

```
for( const auto &[name, value] : gafferAttributes->members() )
{
    ...
```

It makes the rest of the loop a fair bit more readable...